addons/web/static/src/views/list/list_renderer.js

```diff
@@ -1478,6 +1509,193 @@ export class ListRenderer extends Component {
         }
     }
 
+   /**
+    * Create a list of virtual fields for the different property definitions of the 'properties' field
+    *
+    * @param {object} baseFieldObject The 'properties' field
+    * @param {array} propertyDefinitions relevant data to generate property 'fields'
+    * @returns {object} a list of field for the property values
+    */
+    getVirtualPropertyFields(baseFieldObject, propertyDefinitions) {
+        const fields = [];
+        for (const property of propertyDefinitions) {
+            const propertyName = this.getPropertyFieldName(property.name, baseFieldObject.name);
+            fields.push({
+                ...baseFieldObject,
+                FieldComponent: PropertyValue,
+                rawAttrs: {
+                    ...baseFieldObject.rawAttrs,
+                    name: propertyName,
+                },
+                name: propertyName,
+                label: property.string,
+                propertyType: property.type,
+                string: property.string,
+                propertyParentId: property.parentId, //used to group related columns
+                propertyParentName: property.parentName,
+            });
+        }
+        return fields;
+    }
+
+    /**
+     * Get the columns defined for the records with additional virtual columns
+     * Create for the different property values of the 'properties' field
+     *
+     * @param {Array} columns all the columns (typically props.archInfo.columns)
+     * @param {Array} propertyDefinitions relevant data to generate property 'fields'
+     * @returns {Array} the same array with the 'properties' columns replaced by 1 column per property
+     */
+    propertiesGetAllColumns(columns, propertyDefinitions) {
+        columns = [...columns];
+        const propertiesColumnIndices = [];
+        const propertiesFieldNames = Object.keys(propertyDefinitions);
+        for (let i = 0; i < columns.length; i++) {
+            const column = columns[i];
+            if (propertiesFieldNames.includes(column.name)) {
+                propertiesColumnIndices.push(i);
+            }
+        }
+        //indices are assumed sorted
+        let rightOffset = 0;
+        for (const columnIndex of propertiesColumnIndices) {
+            const correctedColumnIndex = columnIndex+rightOffset
+            const baseColumnObject = columns[correctedColumnIndex];
+            const propertyColumns = this.getVirtualPropertyFields(baseColumnObject, propertyDefinitions[baseColumnObject.name]);
+            for (let i = 0; i < propertyColumns.length; i++) {
+                propertyColumns[i].id = `${columnIndex}_property_${i}`;
+            }
+            columns.splice(correctedColumnIndex, 1, ...propertyColumns);
+            rightOffset = rightOffset - 1 + propertyColumns.length;
+        }
+        return columns;
```

Comment:
By trying to optimize this one, I discovered a super cool function in JS, `flatMap` which allow us to do this kind of stuff, tell me what you think :) (filter + map + insert column, all at the same time)

```js
const propertiesFieldNames = Object.keys(propertyDefinitions);

return columns.flatMap((column, columnIndex) => {
    if (!propertiesFieldNames.includes(column.name)) {
        return [column];
    }

    const propertyColumns = this.getVirtualPropertyFields(column, propertyDefinitions[column.name]);
    for (let i = 0; i < propertyColumns.length; i++) {
        propertyColumns[i].id = `${columnIndex}_property_${i}`;
    }
    return propertyColumns;
});
```

Reply:
Oh nice! We get the copy for free too this way :)
Might reuse this a lot.

Not sure why I spliced at all though, could just have built the copy from scratch from the start..